utilities/post_migration.py

```
@@ -214,6 +215,94 @@ def _read_marker() -> str | None:
+                    LOGGER.warning(f"SSH connection not established for VM {vm_name} - retrying...")
+                    return None
+                executor = ssh_conn.rrmngmnt_host.executor(user=ssh_conn.rrmngmnt_user)
+                executor.port = ssh_conn.local_port
+                rc, stdout, err = executor.run_cmd(["lsblk", "-J", "-f"])
+                if rc != 0:
+                    LOGGER.warning(f"lsblk failed on VM {vm_name}: {err} - retrying...")
+                    return None
+                try:
+                    lsblk_data: dict[str, Any] = json.loads(stdout)
+                except json.JSONDecodeError as e:
+                    LOGGER.warning(f"Invalid lsblk JSON on VM {vm_name}: {e} - retrying...")
```

Ruff (0.15.15)

[warning] Logging statement uses f-string

(G004)